### ISSUE TITLE
Sub-workflow arguments with default lead to workflow that errors on run

### ISSUE THREAD

**Issue by jtratner** [dnanexus/dxWDL#284]
I *think* when you have a sub-workflow that's compiled into a fragment with a non-optional argument, with a default, that's not passed in, dxWDL will compile something incorrect. (possibly only if it compiles it into some kind of fragment between two lanes)

For example, we have something like this:

```
    # We put this task here to ensure that we get nice clean "process_lane1"
    # and "process_lane2" steps that do not depend on each other.
    # Otherwise, dxWDL compiles in the `select_all(bam_lane1)` and
    # `select_all(bam_lane2)` *into* the process_lane1 task.  This means that
    # `process_lane2` will not kick off until `process_lane1` completes
    # (because it has a dep on the output of those select_alls).
    call noop as noop_bams_by_lane {
        input: in=lane2_bams[0]
    }
    # by making explicit calls, these are compiled into separate workflow
    # executions, meaning that they will succeed or fail independently. Doing a
    # scatter/gather would make them into subjobs on DNAnexus and thus be harder to debug.
    call lane.lane as process_lane1 {
    <...>
    }

task noop {
input {
   File in
}
command <<<
>>>
}
```

where lane.wdl looks like:

```
workflow lane {
    input {
        some_bool = True
    }
}
```

the compiled workflow then fails with an error about not being able to find `some_bool`'s value.  Also for some reason the `noop` task is compiled into the process_lane1 fragment.

(sent you complete example via email)

**Comment by orodeh** [comment]
I suspect this was fixed in version 1.12.

**Comment by jtratner** [comment]
I hit this with 1.14 (docker version) but I will try again.

**Comment by orodeh** [comment]
@jtratner, I just released version https://github.com/dnanexus/dxWDL/releases/tag/1.15, that should allow you to compile on projects with a large number of applets and workflows. Let's see if we can use it to make progress with this problem.

**Comment by orodeh** [comment]
@jtratner  True. I found a small program that reproduces the issue. The problem occurs when a sub-workflow is called from a fragment. In the example below, calling `Foo` requires providing compulsory argument `a` where `b` already has a default. The bug causes the default for `b` to be lost, so the call is providing only `a`, and not `b`.  

```wdl
version 1.0

import "Foo.wdl" as lib

workflow CounsylBug {
    call Inc { input: a = 10 }
    Int a = 3
    call lib.Foo as Foo { input: a = a }
}

task Inc {
    input {
        Int a
    }
    command {}
    output {
        Int result = a + 1
    }
}
```

```wdl
version 1.0

workflow Foo {
    input {
        Int a
        Int b = 10
    }
    output {
        Int result = a + 1
    }
}
```

**Comment by orodeh** [comment]
I believe I fixed this in release [1.16](https://github.com/dnanexus/dxWDL/releases/tag/1.16). Could you check it out please?

**Comment by jtratner** [comment]
yep fixed it! Thanks :)
